Source/WebCore/page/scrolling/ScrollingStateTree.cpp

```diff
@@ -67,8 +67,37 @@ ScrollingStateTree::~ScrollingStateTree() = default;
 
 void ScrollingStateTree::attachDeserializedNodes()
 {
-    if (m_rootStateNode)
+    // This needs to be done after the move constructor has been called because we deserialize
+    // into a ScrollingStateTree then move to a std::unique_ptr<ScrollingStateTree> and if we
+    // did this in setDeserializedRootStateNode it would be setting nodes' tree pointers to the
+    // wrong ScrollingStateTree.
+    if (m_rootStateNode) {
         m_rootStateNode->attachAfterDeserialization(*this);
+        ASSERT(m_rootStateNode->parentPointersAreCorrect());
+    }
+}
+
+bool ScrollingStateTree::setDeserializedRootStateNode(RefPtr<ScrollingStateFrameScrollingNode>&& node)
```

Comment:
The "Deserialized" bit is really a WK2 implementation detail which is a bit weird to see here. Can we pick a more generic name? Maybe `setRootStateNodeAfterReconstruction` or something.